cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenSecureFields.js

```diff
@@ -56,19 +101,96 @@
         });
     };
 
+    $('button[value="submit-shipping"]').on('click', function (e) {
+        displayPaymentMethods();
+    });
+
+    function displayPaymentMethods() {
+        $('#paymentMethodsUl').empty();
+        if ($('#directoryLookup').val() == 'true') {
+            getPaymentMethods(function (data) {
+                jQuery.each(data.AdyenHppPaymentMethods, function (i, method) {
+                    addPaymentMethod(method, data.ImagePath, data.AdyenDescriptions[i].description);
+                });
+
+                $('input[type=radio][name=brandCode]').change(function () {
+                    $('.hppAdditionalFields').hide();
+                    $('#extraFields_' + $(this).val()).show();
+                });
+            });
+        }
+    };
+
+    function getPaymentMethods(paymentMethods) {
+        $.ajax({
+            url: 'Adyen-GetPaymentMethods',
+            type: 'get',
+            success: function (data) {
+                paymentMethods(data);
+            }
+        });
+    };
+
+    function addPaymentMethod(paymentMethod, imagePath, description) {
+        var li = $('<li>').addClass('paymentMethod');
+        li.append($('<input>')
+            .attr('id', 'rb_' + paymentMethod.name)
+            .attr('type', 'radio')
+            .attr('name', 'brandCode')
+            .attr('value', paymentMethod.brandCode));
+        li.append($('<img>').addClass('paymentMethod_img').attr('src', imagePath + paymentMethod.brandCode + '.png'));
+        li.append($('<label>').text(paymentMethod.name).attr('for', 'rb_' + paymentMethod.name));
+        li.append($('<p>').text(description));
+
+        var additionalFields = $('<div>').addClass('hppAdditionalFields')
+            .attr('id', 'extraFields_' + paymentMethod.brandCode)
+            .attr('style', 'display:none');
+
+        if (paymentMethod.issuers) {
+            var issuers = $('<select>').attr('id', 'issuerList').attr('name', 'issuerId');
+            jQuery.each(paymentMethod.issuers, function (i, issuer) {
+                var issuer = $('<option>')
+                    .attr('label', issuer.name)
+                    .attr('value', issuer.issuerId);
+                issuers.append(issuer);
+            });
+            additionalFields.append(issuers);
+            li.append(additionalFields);
+        }
+        if ($('#OpenInvoiceWhiteList').val().indexOf(paymentMethod.brandCode) !== -1) {
+            // Display Additional Open Invoice fields
+            li.append(additionalFields);
+        }
+        $('#paymentMethodsUl').append(li);
+    };
+
 
     $('button[value="submit-payment"]').on('click', function (e) {
-        if($('#selectedPaymentOption').val() == 'CREDIT_CARD') {
+        if($('#selectedPaymentOption').val() == 'CREDIT_CARD' && $('.payment-information').data('is-new-payment')) {
```

Comment:
Shouldn't this be an id instead of a class and more likely a hidden input?

Reply:
No, this is default SFRA code to identify if it is a new credit card or a stored credit card